Port Ember Version Modal & Minor Changes

richard-cox opened the pull request:
Port the Ember Version Modal
- Previously this was reached via clicking on the version text in the footer, now it's reached via the user menu top right
- This also previously was a modal whereas now it's a page as per other user menu items
- At the moment it looks pretty stark. As it's a page it lacks the modal header with the snazzy logo 
- There's still the same list of versions, the ui one might be handy for QA
- Addresses #2903


Minor Changes
- i10n - User menu text
- Fix - Vertical size of side menu's group menu down arrow now fills menu hover area (no more thin hover colour below down arrow when in dark mode) 
  ![image](https://user-images.githubusercontent.com/18697775/117314518-c0482d00-ae7e-11eb-8f3a-a405722d85f9.png)
  vs 
  ![image](https://user-images.githubusercontent.com/18697775/117314376-9abb2380-ae7e-11eb-9193-a6383e383a2b.png)

- Fix - Cluster Dashboard - Hide metrics Tabbed element if there are no Tabs (no more empty faded square at bottom when no metrics)
- Update support links
  - Home Page - Commercial Support - Link now goes to support page
  - Support Page - Contains links to both rancher support info and contact pages

lvuch commented:
All changes and links lgtm, I'll think about changes to the about page just for visual polish.

@nwmac should we have the logo in the header link to something? home?

vincent99 commented:
Home is finally a single universally-accessible place that you could send that... https://github.com/rancher/rancher/issues/2404

lvuch reviewed (approved):
added table styling

vincent99 commented:
I expected to find the about page by clicking the version (in the footer, or in the slide in menu, not in the User menu (which is otherwise all about me, not Rancher).

The thing labeled "Dashboard Version" is really the Rancher version, this should be `{{vendor}}` (these names have turned out confusing with all the 1 vs 2 UIs and what are they called vs the product changes...).

I tried to set this up at one time but don't think the "ui version" thing to get the dashboard version from git actually works to ever show you useful information for release builds...

The Image list doesn't seem like it really needs to be a table, but if it's going to be it should line up with the one above it.

Shouldn't the release notes be on this page that's all about this version instead of linking to another page (that I get to from a totally different place?)

The header is bad in dark theme.

![image](https://user-images.githubusercontent.com/753917/117430688-354a4f80-aedd-11eb-8c6e-7718f37aa230.png)

richard-cox commented:
Ok, following reviews I've made some changes
- Removed 'About' link from user menu, changed 'version' link in the slide in side nav to point to the About page
- About page
  - Switched the app name to use getVendor instead of getProduct (roughly `Dashboard` --> `Rancher`)
  - Removed the UI version. I'd like to come back to this at some point, it would be handy for QA to know where the dashboard build came from
  - Fixed visual issues regarding tweaks
  - Release notes still exist in a separate page but are reachable from the About page

nwmac commented on assets/translations/en-us.yaml:
```diff
@@ -1309,7 +1329,7 @@ landing:
   commercial:
     title: Commercial Support
-    body: Learn about commercial support <a href="https://rancher.com/support-maintenance-terms/" target='_blank' rel='noopener nofollow noreferrer'>here</a>.
+    body: Learn about commercial support <a href="/support">here</a>.
```
This needs to be done as a nuxt link - this causes a full UI reload when clicked.

richard-cox replied:
Done

nwmac commented on assets/translations/en-us.yaml:
```diff
@@ -150,6 +154,22 @@ suffix:
+    cli: CLI
+    helm: Helm
+    machine: Machine
+    releaseNotes: 'Release notes can be found <a href="/docs/release-notes">here</a>.'
```
This needs to be done as a nuxt link - this causes a full UI reload when clicked as is.

richard-cox replied:
Done

nwmac reviewed (changes requested):
Looks good - just a couple of issues with links that are paths not routes

nwmac reviewed (approved):
LGTM
